clang/lib/CodeGen/CodeGenFunction.cpp

```diff
@@ -1137,6 +1154,10 @@ void CodeGenFunction::StartFunction(GlobalDecl GD, QualType RetTy,
     Fn->addFnAttrs(FnAttrBuilder);
   }
 
+  if (getLangOpts().SYCLIsDevice && D && checkIfDeclaredInSYCLNamespace(D)) {
```

Comment:
Shouldn't this be done only if flag is passed?

Reply:
added a check